src/test/java/seedu/address/model/person/AddressContainsKeywordsPredicateTest.java

```diff
@@ -0,0 +1,82 @@
+package seedu.address.model.person;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.testutil.PersonBuilder;
+
+public class AddressContainsKeywordsPredicateTest {
+
+    @Test
+    public void equals() {
+        List<String> firstPredicateKeywordList = Collections.singletonList("5, Tampines Street 13, #02-03");
+        List<String> secondPredicateKeywordList = Arrays.asList("Main street", "Dhoby Ghaut Street 11");
+
+        AddressContainsKeywordsPredicate firstPredicate =
+                new AddressContainsKeywordsPredicate(firstPredicateKeywordList);
+        AddressContainsKeywordsPredicate secondPredicate =
+                new AddressContainsKeywordsPredicate(secondPredicateKeywordList);
+
+        // same object -> returns true
+        assertTrue(firstPredicate.equals(firstPredicate));
+
+        // same values -> returns true
+        AddressContainsKeywordsPredicate firstPredicateCopy =
+                new AddressContainsKeywordsPredicate(firstPredicateKeywordList);
+        assertTrue(firstPredicate.equals(firstPredicateCopy));
+
+        // different types -> returns false
+        assertFalse(firstPredicate.equals(1));
+
+        // null -> returns false
+        assertFalse(firstPredicate.equals(null));
+
+        // different person -> returns false
+        assertFalse(firstPredicate.equals(secondPredicate));
+    }
+
+    @Test
+    public void test_addressContainsKeywords_returnsTrue() {
+        // One full-matching keyword
+        AddressContainsKeywordsPredicate predicate =
+                new AddressContainsKeywordsPredicate(Arrays.asList("74 University Town, #04-02"));
+        assertTrue(predicate.test(new PersonBuilder().withAddress("74 University Town, #04-02").build()));
+
+        // One full-matching keyword
+        predicate = new AddressContainsKeywordsPredicate(Arrays.asList("73 Jurong Island, #02-04"));
+        assertTrue(predicate.test(new PersonBuilder().withAddress("73 Jurong Island, #02-04").build()));
+
+        // One partial-matching keyword
+        predicate = new AddressContainsKeywordsPredicate(Arrays.asList("University Town"));
+        assertTrue(predicate.test(new PersonBuilder().withAddress("74 University Town, #04-02").build()));
```

Comment:
Could the constants be abstracted out to the Typical Persons class where there are the example addresses?